src/util/movinginterquartilemean.cpp

```diff
@@ -0,0 +1,88 @@
+#include "movinginterquartilemean.h"
+
+MovingInterquartileMean::MovingInterquartileMean(const unsigned int listMaxSize)
+    : m_iListMaxSize(listMaxSize),
+      m_bChanged(true) {
+}
+
+MovingInterquartileMean::~MovingInterquartileMean() {};
+
+double MovingInterquartileMean::insert(double value) {
+    m_bChanged = true;
+
+    // Insert new value
+    if (m_list.isEmpty()) {
+        m_list.prepend(value);
+        m_queue.enqueue(m_list.begin());
+    } else if (value < m_list.first()) {
+        m_list.prepend(value);
+        m_queue.enqueue(m_list.begin());
+    } else if (value >= m_list.last()) {
+        m_list.append(value);
+        m_queue.enqueue(--m_list.end());
+    } else {
+        QLinkedList<double>::iterator i = m_list.begin()++;
+        while (value >= *i) ++i;
+        m_queue.enqueue(m_list.insert(i, value));
+        // (If value already exists in the list, the new instance
+        // is appended next to the old ones: 2·-> 1 2 3 = 1 2 2· 3)
+    }
+    
+    // If list is already full, first delete the oldest value:
+    if (m_list.size() == m_iListMaxSize + 1) {
+        m_list.erase(m_queue.dequeue());
+    }
+    return mean();
+}
+
+void MovingInterquartileMean::clear() {
+    m_bChanged = true;
+    m_queue.clear();
+    m_list.clear();
+}
+
+double MovingInterquartileMean::mean() {
+    if (m_bChanged) {
+        m_bChanged = false;
+        if (m_list.size() <= 4) {
+            double d_sum = 0;
+            foreach (double d, m_list) {
+                d_sum += d;
+            }
+            m_dMean = d_sum / m_list.size();
+        } else if (m_list.size() % 4 == 0) {
+            int quartileSize = m_list.size() / 4;
+            double interQuartileRange = 2 * quartileSize;
+            QLinkedList<double>::iterator i = m_list.begin() + quartileSize;
+            double d_sum = 0;
+            for (int k = 0; k < 2 * quartileSize; ++k) {
```

Comment:
you can iterate over two variables at the same time. It would be better here to do that instead of the loop body. This makes the code a bit more verbose.

```for(int k=0, i=m_list.begin()+quartileSize; k<2*quartileSize; ++i, ++k ) {
d_sum += *i;
```

and maybe change `i` to `el` or 'it'. Single letter variables usually indicate that they are used as an index. But iterators aren't an index.